doc/source/reference/arrays.classes.rst

```diff
@@ -130,7 +172,10 @@ NumPy provides several hooks that classes can customize:
    the subclass and update metadata before returning the array to the
    ufunc for computation.
 
-.. method:: class.__array_wrap__(array, context=None)
+   .. note:: It is hoped to eventually deprecate this method in favour of
+             :func:`__array_ufunc__`.
```

Comment:
But we use `__array_prepare__` in non-ufunc contexts too (albeit a little less often)